src/UI/Component/Table/Factory.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/* Copyright (c) 2017 Nils Haagen <nhaageng@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\Table;
+
+/**
+ * Table factory
+ */
+interface Factory {
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *       The Presentation Table lists some data from the system in a pleasant way.
+	 *       The user should be able to get a quick overview over records in a
+	 *       dataset as simple as possible; for this, only most relevant fields of
+	 *       a record are being displayed at first glance.
+	 *       Rows can be expanded to show more extensive fields and data
+	 *       less important for the identification of a record.
+	 *       The purpose of the presentation and exploration is known, and
+	 *       single records may only be explored in this way,
+	 *   composition: >
+	 *       The Presentation Table consists of a title, a slot for View Controls
+	 *       and Presentation Rows.
+	 *       Rows again will be prefixed by an expander-button and consist of a
+	 *       title, a subtitle and a choice of record-fields.
+	 *       The expanded row will show a lists of further fields and, optionally, buttons.
+	 *   effect: >
+	 *       Rows can be expanded and collapsed to show/hide more extensive and
+	 *       detailed information per record.
+	 *       A click on the expander will enlarge the row vertically to
+	 *       show the complete record. Fields that were shown in the collapsed row will be
+	 *       hidden except for title and subtitle.
+	 *       The ordering or the contents of the table itself can be adjusted with view controls.
+	 *   rivals:
+	 *     1: >
+	 *       Data Table: A data-table shows some dataset and offers tools to
+	 *       explore it in a user defined way. Instead of aiming at simplicity
+	 *       it aims at maximum explorability.
+	 *     2: >
+	 *       Listing Panel: Items represent system-entities; this is not necessarily
+	 *       the case for Presentation Tables.
```

Comment:
Please go into more detail here (as @klees did in his comments). What are system-entities? Why are they different from record? When should I use this control, when Listing Panels. Also add this item here as rival to the Listing Panel and add some short explanation there.

Also, the following explanation was helfpful for me:

> What is crucial to that word entity to me is that the thing it references is persistent, has an identity and (in the best case) also is a member of the domain that is programmatically represented. What entity clearly not means to me is some random set of data from the system (like e.g. some statistic, some collection of answers to a survey or a question, ...).

However, maybe break that down somewhat in more non-technical terms if possible.